build.go

```diff
@@ -34,46 +34,123 @@ import (
+	// add buildpacks to a build.
+	Registry string
+
+	// path to application bits, defaults to current working directory
+	AppPath string
+
+	// Specify the run image.
+	RunImage string
+
+	// If Run Image is empty, it is set to the the 'best' mirror using Builder metadata and AdditionalMirrors.
+	// where 'best' is defined as:
+	//  - if the builder or Additional mirrors has a registry that matches the registry in
+	//    Image, it is 'best'
+	//  - otherwise if there are no matches, use the builder metadata
+	AdditionalMirrors map[string][]string
+
+	// User provided environment variables to the buildpacks,
+	// buildpacks may both read and overwrite these values.
+	Env map[string]string
+
+	// option passed to lifecycle,
+	// publishes Image directly to a remote registry.
+	Publish bool
+
+	// Clear the build cache from previous builds.
+	ClearCache bool
+
+	// TrustBuild when true optimizes builds by running
+	// all lifecycle phases in a single container,
+	// this places registry credentials on the builder's build image.
+	// Only trust builders from reputable sources.
+	TrustBuilder bool
+
+	// List of buildpack images or archives to add to a builder.
+	// These buildpacks may overwrite those on the builder if they
+	// share both an ID and Version with a buildpack on the builder.
+	Buildpacks []string
+
+	// Configure the proxy environment variables,
+	// These variables will only be set in the build image
+	// and will not be used if proxy env vars are already
+	// set.
+	ProxyConfig *ProxyConfig
+
+	// Configure network and volume mounts for the build containers
+	ContainerConfig ContainerConfig
+
+	// Process type that will be used when setting container start command.
 	DefaultProcessType string
-	FileFilter         func(string) bool
-	PullPolicy         config.PullPolicy
+
+	// Filter files from the application source.
+	// If true include file, otherwise exclude
+	FileFilter func(string) bool
+
+	// Strategy for updating images before a build
+	PullPolicy config.PullPolicy
 }
 
+// ProxyConfig specifies proxy setting to be set as environment variables in a container
 type ProxyConfig struct {
-	HTTPProxy  string
-	HTTPSProxy string
-	NoProxy    string
+	HTTPProxy  string // Used to set HTTP_PROXY env var.
+	HTTPSProxy string // Used to set HTTPS_PROXY env var.
+	NoProxy    string // Used to set NO_PROXY env var.
 }
 
+// Additional configuration of the docker container that all build steps
+// occur within.
 type ContainerConfig struct {
+	// Configure network settings of the build container
+	// this string value is handed directly to the docker client,
+	// for valid values of this field see:
+	// https://docs.docker.com/network/#network-drivers
 	Network string
+
+	// Volumes that mounted and accessible during detect & build phases
+	// should have the form: /path/in/host:/path/in/container
+	// for more about volume mounts, and their permissions see
+	// https://docs.docker.com/storage/volumes/
```

Comment:
Maybe it's also worth noting which directories are "sensitive" from a CNB perspective.

Reply:
Think I got the important ones here, but guessing there might be more.